DOCS: refactor developer docs

choldgraf opened the pull request:
This refactors and updates our developer documentation, with the following major changes:

- Information is now correct and updated according to our latest practices (closes #521)
- Splits the contributing docs into a few pages, so it's not as cluttered (also creates a new subdirectory `contribute/` and adds a redirect)
- Creates a "getting started" page that has the minimal information needed to make edits etc
- Creates a separate page for installing the dev environment manually, for more advanced users
- Creates a more extensible "topic guides" page for little how-tos and tips
- Converts the rST into MyST Markdown for this content
- Cleans up some filesystem and configuration cruft to make it easier to edit the docs

damianavila commented on docs/contribute/setup.md:
```diff
@@ -0,0 +1,150 @@
+This will install the necessary dependencies and build the documentation located in the `docs/` folder.
+They will be placed in a `docs/_build/html` folder.
+If the docs have already been built, it will only build new pages that have been updated.
+You can open one of the HTML files there to preview the documentation locally.
```
Do we want/need to mention `http.server` to serve that content easily and navigate into the pages?

choldgraf replied:
that's a cool idea - I can't remember the command, wanna add it?

damianavila replied:
https://github.com/choldgraf/pydata-sphinx-theme/pull/4

damianavila commented on docs/contribute/topics.md:
```diff
@@ -0,0 +1,192 @@
+### Fix accessibility errors
+
+Start by checking for issues on the
+[accessibility roadmap](https://github.com/pandas-dev/pydata-sphinx-theme/blob/master/docs/a11y-roadmap.txt).
```
```suggestion
[accessibility roadmap](https://github.com/pydata/pydata-sphinx-theme/blob/master/docs/a11y-roadmap.txt).
```

damianavila commented on noxfile.py:
```diff
@@ -1,22 +1,53 @@
+    force_reinstall = "reinstall" in session.posargs
+    should_install = not sphinx_is_installed or force_reinstall
+    if should_install:
+        session.log("Installing fresh environment...")
```
Don't you need to return `True` somehow so it executes the block under the `if`?
Or am I missing something obvious because is Friday evening :wink:?

choldgraf replied:
+1

choldgraf commented on noxfile.py:
```diff
@@ -1,22 +1,53 @@
+    """
+    bin_files = list(Path(session.bin).glob("*"))
+    sphinx_is_installed = any("sphinx-build" in ii.name for ii in bin_files)
+    force_reinstall = "reinstall" in session.posargs
+    should_install = not sphinx_is_installed or force_reinstall
+    if should_install:
+        session.log("Installing fresh environment...")
+    else:
+        session.log("Skipping environment install...")
+
+
```
```suggestion
    force_reinstall = "reinstall" in session.posargs
    should_install = not sphinx_is_installed or force_reinstall
    if should_install:
        session.log("Installing fresh environment...")
    else:
        session.log("Skipping environment install...")
    return should_install


```

damianavila reviewed (approved):
LGTM after the CI is green!

jorisvandenbossche reviewed (commented):
This is a _really_ nice improvement of the contributing docs, thanks a lot @choldgraf !

I did a read through and added some comments / questions.

choldgraf commented on docs/conf.py:
```diff
@@ -64,11 +65,6 @@
-html_sidebars = {
-    "contributing": ["search-field", "custom-template"],
-    "changelog": [],
-}
```
Hmm good point - I think it's reasonable to leave in here. Let's leave the `custom-template` in there, and I'll add a test for what an empty sidebar looks like into https://github.com/pydata/pydata-sphinx-theme/pull/544

choldgraf commented on docs/contribute/index.md:
```diff
@@ -0,0 +1,58 @@
+
+## Location and structure of CSS/JS assets
+
+The CSS and JS for this theme are built for the browser from `src/*` with
```
good catch - fixed in next commit

choldgraf commented on docs/contribute/manual.md:
```diff
@@ -0,0 +1,75 @@
+
+# Set up a manual development environment
+
+If you prefer not to use automation tools like `nox`, you may also manually set up a development environment locally.
```
added 👍

choldgraf commented on docs/contribute/setup.md:
````diff
@@ -0,0 +1,158 @@
+$ pre-commit install
+```
+
+This will install the necessary dependencies to run `pre-commit` every time you make a commit with Git.
````
ah no - they will be in your local interpreter environnment. I'll add a note that clarifies this

jorisvandenbossche replied:
Ah, I always thought pre-commit also used some kind of environment (see eg https://stackoverflow.com/questions/62539288/where-does-pre-commit-install-environments)

choldgraf commented on docs/contribute/topics.md:
```diff
@@ -0,0 +1,192 @@
+$ nox --force-venv-backend none
+
+# alternatively:
+$ nox -fb none
```
ah I think that's a more intuitive shortcut, I'll replace the current one with your suggestion

choldgraf commented on noxfile.py:
```diff
@@ -1,22 +1,54 @@
+    We assume that if `sphinx-build` is in the bin/ path, the environment is
+    installed.
+    """
+    bin_files = list(Path(session.bin).glob("*"))
```
Good catch - I'll add in a check for `--no-venv` in the next commit so it still works for that case

choldgraf commented on noxfile.py:
```diff
@@ -1,22 +1,54 @@
+@nox.session
+def compile(session):
+    if _should_install(session):
+        session.install(".")
```
I think that we do need it, because this is what gives us the theme builder **CLI**, which is not bundle with it by default via `install(".")`. Is that what you were asking?

choldgraf commented:
OK I believe I've addressed all of your suggestions and comments @jorisvandenbossche - I've "resolved" them via the github UI, let me know if I missed something!

jorisvandenbossche commented on docs/conf.py:
```diff
@@ -107,6 +107,13 @@
 }
 
+html_sidebars = {
+    "contributing": [
```
The contributing top-level page/directory doesn't exist anymore, so this has no effect?

choldgraf replied:
good point - I updated this in the latest commit

jorisvandenbossche commented on docs/conf.py:
```diff
@@ -107,6 +107,14 @@
 }
 
+html_sidebars = {
+    "contributing/index": [
```
```suggestion
    "contribute/index": [
```

jorisvandenbossche commented:
Thanks a lot @choldgraf !
